### ISSUE TITLE
Find Substitute for Google Authenticator Class

### ISSUE THREAD

**Issue by therealssj** [d8-contrib-modules/tfa#35]
**Motivattion**
- The google authenticator class saw it's last commit about a year ago and is not maintained anymore.
- There are quite a few pull requests related to security mainly targetting timing attacks which we should definitely have in the class we use.

**Resolution**
1. Find a different repo which is much more updated than the current one.
2. Fork the Google Authenticator repo and update it as needed.

**Comment by therealssj** [comment]
This repo https://github.com/ChristianRiesen/otp looks good.
It doesn't have recovery codes yet but I created an issue and the author replied that he can add that as a convenience function.

**Comment by nerdstein** [comment]
Awesome!

**Comment by nerdstein** [comment]
Let's give it a shot. Is it on composer/packagist?

**Comment by therealssj** [comment]
Yes it is. I will add it soon. It is still missing recovery codes so I am waiting on the author to add that in.
I went through the code once and it is good :)

**Comment by therealssj** [comment]
The library now supports recovery codes and I recently made a pull request to fix another issue.

**Comment by therealssj** [comment]
Using https://github.com/ChristianRiesen/otp now
